🐳 Added docker & compose

johnsonjo4531 opened the pull request:
Added Docker and compose according to issue https://github.com/mekb-turtle/discord-ai-bot/issues/11

There should be enough documentation in the new readme and comments in the new Makefile to show you what's happening. Let me know if I can provide any future updates to this or if you have any questions. Thanks for this project it's pretty awesome!

johnsonjo4531 committed:
Added docker & compose 🐳

sophuric commented on README.md:
Could you instead write something like "repeat steps 3-7 from above instructions" so we aren't repeating the same instructions?

johnsonjo4531 replied:
Yes that makes perfect sense :D I was merely worried about the step order ever changing, but that seems easier to remember and update than repeating the steps.

johnsonjo4531 replied:
This should be resolved in https://github.com/mekb-turtle/discord-ai-bot/pull/13/commits/e7e087d87819f01ebda8d8b1a098bc4d4ad19451

sophuric commented on README.md:
```diff
@@ -31,3 +31,31 @@
+### Set-up instructions Docker
+1. Install [docker](https://docs.docker.com/get-docker/)
+   - should be atleast compatible with version 3 of compose (docker engine 1.13.0+)
+2. Install [Ollama](https://github.com/jmorganca/ollama) (ditto)
```
ditto means same thing as before, doesn't make sense here, put "if you have a package manager, use that instead to install this"

johnsonjo4531 replied:
Sounds good and good catch :+1:

johnsonjo4531 replied:
This should be resolved in https://github.com/mekb-turtle/discord-ai-bot/pull/13/commits/e7e087d87819f01ebda8d8b1a098bc4d4ad19451

sophuric commented on .editorconfig:
Why is .editorconfig here if we have .eslintrc.json? or is this for yaml and other files?

johnsonjo4531 replied:
Yeah it's mainly for the yaml, but I'm not even sure if it was working properly for it. I could just leave the spaces hard coded and delete this file.

johnsonjo4531 replied:
This should be resolved in https://github.com/mekb-turtle/discord-ai-bot/pull/13/commits/e7e087d87819f01ebda8d8b1a098bc4d4ad19451

sophuric commented on Dockerfile:
```diff
@@ -0,0 +1,7 @@
+RUN npm i --omit=dev --no-package-lock
+USER node
+
+CMD ["node","./src/index.js"]
```
what about `npm start`?

johnsonjo4531 replied:
Good question. I merely did it this way, because I saw this in the Docker best-practices for Node.js. I can switch it to npm start though if you're fine with the below.

> CMD
> When creating an image, you can bypass the package.json's start command and bake it directly into the image itself. First off this reduces the number of processes running inside of your container. Secondly it causes exit signals such as SIGTERM and SIGINT to be received by the Node.js process instead of npm swallowing them.
>
> ```Dockerfile
> CMD ["node","index.js"]
> ```
> https://github.com/nodejs/docker-node/blob/main/docs/BestPractices.md#cmd

johnsonjo4531 replied:
I'm awaiting further input before I make the requested change  :+1: (just wanted to make sure you knew that this was **not yet** resolved in https://github.com/mekb-turtle/discord-ai-bot/pull/13/commits/e7e087d87819f01ebda8d8b1a098bc4d4ad19451)

johnsonjo4531 committed:
✏️📄 Making requested edits

Updating readme removing .editorconfig
according to the requested changes
in pr https://github.com/mekb-turtle/discord-ai-bot/pull/13

sophuric commented:
lgtm, i'm away for a week so i might not have time right now, but i'll test it after and let you know

johnsonjo4531 commented:
@mekb-turtle I can wait a week for sure. Feel free to ask me any questions if you can't get it running after you are back.

johnsonjo4531 committed:
Fixing issue with .env.example

See https://github.com/mekb-turtle/discord-ai-bot/issues/11#issuecomment-1888417710 for the issue. I was also having a similar issue with this .env.example file when running in docker and had to make the fix in my .env.

sophuric commented:
lgtm, sorry it took so long, i was busy with other stuff
